### PR TITLE
Test client - upload_activity

### PR DESCRIPTION
**Pull request by jsamoocha**
Fully tests upload_activity(). (part of #250)

I could use some advice on if/how to use `FutureWarning`s in case of seemingly deprecated API parameters.

### CODE REVIEW

**Comment by lwasser** [comment]
oh @jsamoocha i can ask some of my colleagues about what the best practices are for this. but question - are we dropping those API elements now? (i am asking to understand better how future warning works). Is future warning for an instance where we might drop this in the next release vs it no longer exists now? 

Also it looks like this PR has some merge conflicts

**Comment by lwasser** [comment]
or maybe @hozn has thoughts on how we handle deprecated api endpoints? it seems like we are dropping them now

File: stravalib/client.py [hunk 1]
```diff
@@ -686,13 +683,24 @@ def upload_activity(self, activity_file, data_type, name=None, description=None,
                               nordicski, alpineski, backcountryski, iceskate, inlineskate,
                               kitesurf, rollerski, windsurf, workout, snowboard, snowshoe
                               Type detected from file overrides, uses athlete's default type if not specified
+                              WARNING - This param is supported (as of 2022-11-15), but not
```
Comment (lwasser) [inline stravalib/client.py:686]: @jsamoocha so is this not documented in the strava API docs?? or in our docs?  many thanks

File: stravalib/client.py [hunk 2]
```diff
@@ -686,13 +683,24 @@ def upload_activity(self, activity_file, data_type, name=None, description=None,
...
+                              documented and may be removed in the future.
         :type activity_type: str
 
-        :param private: (optional) set to True to mark the resulting activity as private, 'view_private' permissions will be necessary to view the activity
+        :param private: (optional) set to True to mark the resulting activity as private,
+                        'view_private' permissions will be necessary to view the activity.
```
Comment (lwasser) [inline stravalib/client.py:691]: similar question to above - if it's not supported in the strava API does that mean it just no longer works? so we should remove it and add a warning to the user that this particular API feature has been removed by strava?

**Comment by jsamoocha** [comment]
> oh @jsamoocha i can ask some of my colleagues about what the best practices are for this. but question - are we dropping those API elements now? (i am asking to understand better how future warning works). Is future warning for an instance where we might drop this in the next release vs it no longer exists now?
> 
> Also it looks like this PR has some merge conflicts

In this PR, no API elements are dropped from stravalib.

I'm not in favor of removing deprecated request parameters as long as Strava doesn't send an error response because of them (I manually checked on the actual Strava API with a test account). Dropping them may break code that has dependencies on stravalib. So instead, by using [`warnings`](https://docs.python.org/3/library/warnings.html) (and logging them), users using the deprecated parameters will have time to address these issues while their code remains functioning.

I was mainly wondering about the best way to import/configure warnings. I did everything on the module level (i.e., in `client.py`), but maybe there are better ways?

File: stravalib/client.py (hunk 1 above)
Comment (jsamoocha) [reply stravalib/client.py:686]: `activity_type` doesn't officially exist anymore as a request parameter per the Strava API docs. However, it still seems to be functioning - I manually checked that providing this parameter makes strava assign the type to the uploaded activity.

File: stravalib/client.py (hunk 2 above)
Comment (jsamoocha) [reply stravalib/client.py:691]: `private` also doesn't exist anymore per the Strava API docs. But contrary to `activity_type`, it seems to be ignored by Strava: providing private=True/"true"/1/etc doesn't make the uploaded activity private.

**Review by yihong0618 — APPROVED** [review]
LGTM

**Comment by hozn** [comment]
> or maybe @hozn has thoughts on how we handle deprecated api endpoints? it seems like we are dropping them now

If the API endpoints are simply deprecated, my general thought would be:
- Emit a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) from the method in question for the next point or even patch-level release.
- Remove the method in the next major release.  (Which may mean forcing cut of major release due to Strava dropping the method.)

**Pull request merged**